### PR TITLE
wdio - clipping for native fps

### PR DESCRIPTION
**Pull request by paweltomaszewskisaucelabs**
wdio - clipping for native fps

### CODE REVIEW

File: visual-js/visual-wdio/src/SauceVisualService.ts [hunk 1]
```diff
@@ -402,16 +402,18 @@ export default class SauceVisualService implements Services.ServiceInstance {
       );
 
       const clipSelector = options.clipSelector ?? this.clipSelector;
-      const clipElement = clipSelector
-        ? await browser.$(clipSelector).elementId
-        : null;
+      const clipElement =
+        clipSelector && !fullPageConfig
```
[the remark this answers is not in the document]
Comment (paweltomaszewskisaucelabs) [reply visual-js/visual-wdio/src/SauceVisualService.ts:406]: to extract `isApp` info from `metaInfo` we will need to repeat some logic from backend like `safelyParseSessionMetadata` so I would prefer avoid it and do it only on backend side

[the remark this answers is not in the document]
Comment (paweltomaszewskisaucelabs) [reply visual-js/visual-wdio/src/SauceVisualService.ts:406]: elements will be find at the backend not in sdk so there will be no difference in behaviour. This is also needed to cliping and ignore selectors for lazy load elements in web (avoid asking users to scroll and load page before taking fps screenshot).
Is it ok for you guys to merge it like this?

[the remark this answers is not in the document]
Comment (FriggaHel) [reply visual-js/visual-wdio/src/SauceVisualService.ts:406]: I'd prefer to have a it in either:
- An element
- A dedicated field for ignore regions as part of FPS config.

I see no reason why we should continue to support non-elementId for non-FPS screenshot. IMO that should get discontinued as soon as not used by users.

For suggestion 2, not sure if we should go in that direction, but input between FPS and non-FPS diverge so much that I wonder if it should even be the same call in the end.

[the remark this answers is not in the document]
Comment (paweltomaszewskisaucelabs) [reply visual-js/visual-wdio/src/SauceVisualService.ts:406]: for non FPS visual check will always search for an element on the sdk.
this pull request has nothing with ignore regions it's only for clipping.

Clipping is fully functional. I don't understand why it shouldn't been merge. It has nothing to with splitting it into separate call.

[the remark this answers is not in the document]
Comment (FriggaHel) [reply visual-js/visual-wdio/src/SauceVisualService.ts:406]: I mixed region / clipping but that does not change the observation.

What was before using `clipElement` (FPS + Browser) will now use `clipSelector` which should not be done IMO.

**Pull request merged**